watertap/unit_models/pressure_exchanger.py

```diff
@@ -352,6 +417,93 @@ def eq_mass_transfer_term(b, t, p, j):
                     == -b.low_pressure_side.mass_transfer_term[t, p, j]
                 )
 
+            if self.config.has_leakage and not self.config.has_mixing:
+
+                @self.Constraint(
+                    self.flowsheet().config.time, doc="High pressure side mass transfer"
+                )
+                def eq_leakage(b, t):
+                    return (
+                        b.high_pressure_side.properties_out[t].flow_vol
+                        == (1 - b.leakage_vol[t])
+                        * b.high_pressure_side.properties_in[t].flow_vol
+                    )
+
+                @self.Constraint(
+                    self.flowsheet().config.time,
+                    self.config.property_package.phase_list,
+                    self.config.property_package.solute_set,
+                    doc="Mixing effect of the unit",
+                )
+                def eq_mixing(b, t, p, j):
+                    return (
+                        b.low_pressure_side.properties_out[t].conc_mass_phase_comp[p, j]
+                        == b.low_pressure_side.properties_in[t].conc_mass_phase_comp[
+                            p, j
+                        ]
+                    )
```

Comment:
I don't think this is correct. If there is some leakage, it will impact the concentration. This implementation essentially would allow leakage which is basically dictated by the water mass transfer term, but then the concentration would be held constant by adjusting the salt mass transfer term.

Maybe the real fix is that we don't allow there to be no mixing if there is leakage (i.e. we return an error if leakage = True and mixing = False)

Reply:
I revise this to directly raise an error if only set `has_leakage=True`.